diff --git a/Makefile b/Makefile
index c254926..d0ee607 100644
--- a/Makefile
+++ b/Makefile
@@ -44,11 +44,9 @@ clean:
 
 module-install:
 	@$(MAKE) -C $(KERNELDIR) M=$(PWD) modules_install
-	depmod -a
 
 install:
 	@$(MAKE) -C $(KERNELDIR) M=$(PWD) modules_install
-	depmod -a
 	@$(MAKE) -C tools install
 
 rwildcard=$(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2) $(filter $(subst *,%,$2),$d))
