diff --git a/lib/packlib.c b/lib/packlib.c
index d0bb181..38f028c 100644
--- a/lib/packlib.c
+++ b/lib/packlib.c
@@ -551,7 +551,7 @@ PWClose(pwp)
 	    PWDICT tmp_pwp;
 
 	    memcpy(&tmp_pwp, pwp, sizeof(PWDICT));
-	    HwmsHostToBigEndian(tmp_pwp.hwms, sizeof(tmp_pwp.hwms), en_is32);
+	    HwmsHostToBigEndian((char*)tmp_pwp.hwms, sizeof(tmp_pwp.hwms), en_is32);
 	    fwrite(tmp_pwp.hwms, 1, sizeof(tmp_pwp.hwms), pwp->wfp);
 	}
     }
