Index: mtools-4.0.19/dirCache.c
===================================================================
--- mtools-4.0.19.orig/dirCache.c
+++ mtools-4.0.19/dirCache.c
@@ -149,9 +149,9 @@ dirCache_t *allocDirCache(Stream_t *Stre
 			return 0;
 		}
 		(*dcp)->nr_entries = (slot+1) * 2;
-		memset( (*dcp)->bm0, 0, DC_BITMAP_SIZE);
-		memset( (*dcp)->bm1, 0, DC_BITMAP_SIZE);
-		memset( (*dcp)->bm2, 0, DC_BITMAP_SIZE);
+		memset( (*dcp)->bm0, 0, sizeof((*dcp)->bm0));
+		memset( (*dcp)->bm1, 0, sizeof((*dcp)->bm1));
+		memset( (*dcp)->bm2, 0, sizeof((*dcp)->bm2));
 		(*dcp)->nrHashed = 0;
 	} else
 		if(growDirCache(*dcp, slot) < 0)
Index: mtools-4.0.19/direntry.c
===================================================================
--- mtools-4.0.19.orig/direntry.c
+++ mtools-4.0.19/direntry.c
@@ -24,6 +24,7 @@
 
 void initializeDirentry(direntry_t *entry, Stream_t *Dir)
 {
+	memset(entry, 0, sizeof(*entry));
 	entry->entry = -1;
 /*	entry->parent = getDirentry(Dir);*/
 	entry->Dir = Dir;
