Index: cacao-1.6.1/src/scripts/java.in
===================================================================
--- cacao-1.6.1.orig/src/scripts/java.in
+++ cacao-1.6.1/src/scripts/java.in
@@ -28,7 +28,7 @@ CURRENT_DIR=`cd "${SH_DIR}" && pwd`
 # if this shell script is invoked through symlink, we get invalid paths. As getting the real path to this file is
 # very painful using only /bin/sh, python is used instead.
 PYTHON_STR="import os; print os.path.dirname(os.path.realpath(os.readlink(\"${CURRENT_DIR}/java\"))) if os.path.islink(\"${CURRENT_DIR}/java\") else \"${CURRENT_DIR}\""
-REAL_DIR=`python -c "${PYTHON_STR}"`
+REAL_DIR=`python2 -c "${PYTHON_STR}"`
 
 PARENT_DIR=`dirname ${REAL_DIR}`
 
