Index: git/Makefile.am
===================================================================
--- git.orig/Makefile.am
+++ git/Makefile.am
@@ -24,7 +24,6 @@ include $(srcdir)/%D%/contrib/Makefile.a
 include $(srcdir)/%D%/etc/Makefile.am
 include $(srcdir)/%D%/lib/Makefile.am
 include $(srcdir)/%D%/man/Makefile.am
-include $(srcdir)/%D%/tests/Makefile.am
 if BUILD_LINUX
 include $(srcdir)/%D%/udev/Makefile.am
 endif
Index: git/contrib/Makefile.am
===================================================================
--- git.orig/contrib/Makefile.am
+++ git/contrib/Makefile.am
@@ -4,8 +4,6 @@ include $(srcdir)/%D%/zcp/Makefile.am
 
 if BUILD_LINUX
 include $(srcdir)/%D%/bpftrace/Makefile.am
-include $(srcdir)/%D%/dracut/Makefile.am
-include $(srcdir)/%D%/initramfs/Makefile.am
 endif
 
 if PAM_ZFS_ENABLED
