Index: xfsprogs-4.14.0/libxfs/Makefile
===================================================================
--- xfsprogs-4.14.0.orig/libxfs/Makefile
+++ xfsprogs-4.14.0/libxfs/Makefile
@@ -135,7 +135,7 @@ crc32table.h: gen_crc32table.c crc32defs
 # disk.
 crc32selftest: gen_crc32table.c crc32table.h crc32.c crc32defs.h
 	@echo "    [TEST]    CRC32"
-	$(Q) $(BUILD_CC) $(BUILD_CFLAGS) -D CRC32_SELFTEST=1 crc32.c -o $@
+	$(Q) $(BUILD_CC) $(BUILD_CFLAGS) -I$(STAGING_INCDIR) -D CRC32_SELFTEST=1 crc32.c -o $@
 	$(Q) ./$@
 
 # set up include/xfs header directory
