diff --git a/SConstruct b/SConstruct
index 5a8a48daff2..2823db174ea 100644
--- a/SConstruct
+++ b/SConstruct
@@ -3520,9 +3520,9 @@ def doConfigure(myenv):
 
         if not conf.CheckLibWithHeader(
                 cryptoLibName,
-                ["openssl/crypto.h"],
+                "openssl/crypto.h",
                 "C",
-                "SSLeay_version(0);",
+                call="SSLeay_version(0);",
                 autoadd=True):
             maybeIssueDarwinSSLAdvice(conf.env)
             conf.env.ConfError("Couldn't find OpenSSL crypto.h header and library")
@@ -3679,7 +3679,7 @@ def doConfigure(myenv):
         if conf.CheckLibWithHeader(
             "curl",
             ["curl/curl.h"], "C",
-            "curl_global_init(0);",
+            call="curl_global_init(0);",
             autoadd=False):
             return True
 
@@ -3805,7 +3805,7 @@ def doConfigure(myenv):
             "sasl2",
             ["stddef.h","sasl/sasl.h"],
             "C",
-            "sasl_version_info(0, 0, 0, 0, 0, 0);",
+            call="sasl_version_info(0, 0, 0, 0, 0, 0);",
             autoadd=False ):
         myenv.ConfError("Couldn't find SASL header/libraries")
 
@@ -3970,7 +3970,7 @@ def doConfigure(myenv):
                 ["mongoc-1.0"],
                 ["mongoc/mongoc.h"],
                 "C",
-                "mongoc_get_major_version();",
+                call="mongoc_get_major_version();",
                 autoadd=False ):
             conf.env['MONGO_HAVE_LIBMONGOC'] = "library"
         if not conf.env['MONGO_HAVE_LIBMONGOC'] and env.TargetOSIs('darwin') and conf.CheckMongoCFramework():
