Index: git/SConstruct
===================================================================
--- git.orig/SConstruct
+++ git/SConstruct
@@ -2217,7 +2217,7 @@ if env.TargetOSIs('posix'):
     env.Append( CCFLAGS=["-fno-omit-frame-pointer",
                          "-fno-strict-aliasing",
                          "-fasynchronous-unwind-tables",
-                         "-ggdb" if not env.TargetOSIs('emscripten') else "-g",
+                         #"-ggdb" if not env.TargetOSIs('emscripten') else "-g", # WILD REMOVE, let yocto define debug flags
                          "-pthread",
                          "-Wall",
                          "-Wsign-compare",
@@ -2643,7 +2643,11 @@ def doConfigure(myenv):
 
         # Warn about redundant moves, such as moving a local variable in a return that is different
         # than the return type.
-        AddToCXXFLAGSIfSupported(myenv, "-Wredundant-move")
+        # WILD: suppress this warning, it's very noisy and makes it hard to find real compile errors.
+        AddToCXXFLAGSIfSupported(myenv, "-Wno-redundant-move")
+        # WILD: also suppress some more noisy warnings
+        AddToCXXFLAGSIfSupported(myenv, "-Wno-c++20-compat")
+        AddToCXXFLAGSIfSupported(myenv, "-Wno-interference-size")
 
         # Disable warning about variables that may not be initialized
         # Failures are triggered in the case of boost::optional in GCC 4.8.x
