From 12310726629fc2460cf62b4d94ac967bf5b58945 Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Mon, 12 Nov 2018 21:40:49 -0500
Subject: [PATCH] add thud to LAYERSERIES_COMPAT

---
 conf/layer.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/layer.conf b/conf/layer.conf
index 8e8a300..4bcf834 100644
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -7,7 +7,7 @@ BBFILES += "${LAYERDIR}/recipes*/*/*.bb ${LAYERDIR}/recipes*/*/*.bbappend"
 BBFILE_COLLECTIONS += "meta-java"
 BBFILE_PATTERN_meta-java := "^${LAYERDIR}/"
 BBFILE_PRIORITY_meta-java = "10"
-LAYERSERIES_COMPAT_meta-java = "sumo"
+LAYERSERIES_COMPAT_meta-java = "sumo thud"
 
 LICENSE_PATH += "${LAYERDIR}/licenses"
 
-- 
2.19.1

